lib/entities/request/index.ts

```diff
@@ -61,6 +63,7 @@ export function parseQuoteRequests(body: QuoteRequestBodyJSON): {
     slippageTolerance: body.slippageTolerance ?? DEFAULT_SLIPPAGE_TOLERANCE,
     swapper: body.swapper,
     sendPortionEnabled: body.sendPortionEnabled,
+    forcePortion: body.forcePortionString === process.env.FORCE_PORTION_STRING,
```

Comment:
This still feels awkward, and I've been thinking an alternative. I propose to get the string only once during injector time https://github.com/Uniswap/unified-routing-api/blob/main/lib/handlers/quote/injector.ts#L70:

```
const forcePortionString: string | undefined = process.env. FORCE_PORTION_STRING;

    return {
      quoters: {
        [RoutingType.DUTCH_LIMIT]: new RfqQuoter(paramApiUrl, serviceUrl, paramApiKey, forcePortionString),
        [RoutingType.CLASSIC]: new RoutingApiQuoter(routingApiUrl, routingApiKey, portionProvider, tokenFetcher, forcePortionString),
      },
      rpcUrlMap,
      tokenFetcher: tokenFetcher,
      permit2Fetcher: new Permit2Fetcher(rpcUrlMap),
      syntheticStatusProvider: new UPASyntheticStatusProvider(paramApiUrl, synthSwitchApiKey),
    };
```

Then inside both `RfqQuoter` and `RoutingApiQuoter`, since you have the access to the `QuoteRequestInfo.forcePortionString`, then you easily pass down `const forcePortion = QuoteRequestInfo.forcePortionString === forcePortionString` boolean flag to `PortionFetcher`

This means you no longer need to have the special `forcePortionString` in `QuoteRequestBodyJSON`, and there's no need to omit `forcePortion` from `QuoteRequestBodyJSON` anymore. Conceptually `forcePortionString` is a shared request field across classic and dutch, so putting it inside `QuoteRequestInfo` makes sense.

Reply:
yea generally agree with the above on placing the logic into the injector, and would prefer not putting it in QuoteRequestBodyJSON in case that gets logged somewhere else.